sympy/solvers/solveset.py

```diff
@@ -255,7 +256,9 @@ def _invert_real(f, g_ys, symbol):
                     res = imageset(root, g_ys_pos)
                     base_positive = solveset(base >= 0, symbol, S.Reals)
                     _inv, _set = _invert_real(base, res, symbol)
-                    return (_inv, _set.intersect(base_positive))
+                    if _inv == symbol:
+                        return (_inv, _set.intersect(base_positive))
+                    return (_inv, _set)
```

Comment:
So if `_invert_real` does not fully invert to symbol then we shouldn't automatically intersect the two sets. That makes sense but does there not need to be something else to handle the kind of case that `base_positive` is intended to handle?

Reply:
Probably, in order to be completely correct, we need to do the intersection with the set `base_positive` transformed with the same transformation the function `_invert_real` maps `base` to `_inv`.
I do not have any idea how to obtain the transformation used inside `_invert_real`.

Reply:
I'm not sure how to implement this either.

The change here is an improvement though so I think we could use this. We should add a comment though clearly indicating that something is not being handled properly.

What would be a test case where not handling the sign of the base properly leads to an incorrect result?

Reply:
Here is the code
```
                if den % 2 == 0 and num % 2 == 1 and den.is_zero is False:
                    root = Lambda(n, real_root(n, expo))
                    g_ys_pos = g_ys & Interval(0, oo)
                    res = imageset(root, g_ys_pos)
                    base_positive = solveset(base >= 0, symbol, S.Reals)
                    _inv, _set = _invert_real(base, res, symbol)
                    if _inv == symbol:
                        return (_inv, _set.intersect(base_positive))
                    return (_inv, _set)
```
We have to invert `f = g_y` and `f` is `Pow` where the exponent has not symbol (i.e. it is a constant).
In particular the exponent is rational with even denominator. In this case `res` is a set of positive numbers (I do not see what is the advantage of using `g_y_pos` instead of `g_y`).
Therefore we want to solve `base = res` using `_invert_real`. So what is the use of `base_positive`, since `res` is already positive?

By the way, I tried to comment out the code containing `base_positive`. The test `test_issue_21022` in `test_solveset.py` is the only one failing. 
The test is quite complicated and I do not know why it fails.

Reply:
That particular test was recently changed in #23011. Did you try with the old or the new version of the test?

Reply:
I tried, but it fails.
The problem is that `len(res)` now is `8` and not `4`. So it seems that `base_positive` needs at some point.

Reply:
On master it works!

Reply:
Yes, that makes sense.